src/core/utils/props.ts

```diff
@@ -60,12 +60,10 @@ export interface PolymorphicForwardRefComponent<
     >
   > {
   <As = T>(
-    props: As extends ''
-      ? { as: keyof JSX.IntrinsicElements }
+    props: As extends keyof JSX.IntrinsicElements
+      ? Merge<JSX.IntrinsicElements[As], OwnProps & { as?: As }>
```

Comment:
Could you explain to us mortals what it does? 😀

Reply:
It's basically the same as before, the difference is that previously it was checking against empty string first to make sure that `as` prop was actually used, whereas now it's _always_ checking against `JSX.IntrinsicElements` first. I'm guessing the empty string was causing problems in itwinjs-core